src/apps/checklist-button/checklist-material-button.entry.jsx

```diff
@@ -62,36 +65,40 @@ function ChecklistMaterialButtonEntry({
     }
   };
 
-  if (status === "pending") {
-    if (loggedIn) {
-      // If we're pending and logged in, then trigger the actual request.
-      dispatch(action({ materialListUrl, materialId: id })).then(() =>
-        dispatch(resetStatus({ materialId: id })).then(() =>
-          setOnList(newOnList)
-        )
-      );
-    } else if (User.authenticationFailed()) {
-      // If authentication failed, abort.
-      dispatch(aborted({ materialId: id }));
+  useEffect(() => {
+    if (status === "pending") {
+      if (loggedIn) {
+        // If we're pending and logged in, then trigger the actual request.
+        dispatch(action({ materialListUrl, materialId: id })).then(payload => {
+          dispatch(resetStatus({ materialId: id })).then(() => {
+            dispatch(
+              setInitialStatus({
+                materialId: id,
+                onList: payload?.error ? onList : newOnList
+              })
+            );
+          });
```

Comment:
There seems to be some similarities between `resetStatus()` and `setInitialStatus()` - partly in naming, partly in parameters. Could we merge the two of them together?